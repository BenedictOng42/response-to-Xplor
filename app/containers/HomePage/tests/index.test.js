import React from 'react';
import { shallow } from 'enzyme';

import { HomePage } from '../index';

describe('<HomePage />', () => {
  const wrapper = shallow(<HomePage />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  const instance = wrapper.instance();
  it('should run #onChange properly', () => {
    const event = {
      target: {
        value: 'word',
      },
    };
    instance.onChange(event);
    expect(instance.state.command).toEqual('word');
  });

  it('should run #onInputKeyDown and onAddCommand if keyCode is 13', () => {
    const keyCode = 13;
    instance.onAddCommand = jest.fn();
    instance.onInputKeyDown(keyCode);
    expect(instance.onAddCommand).toHaveBeenCalledTimes(1);
  });
  it('should run #onInputKeyDown and NOT the onAddCommand if keyCode is not 13', () => {
    const keyCode = 1;
    instance.onAddCommand = jest.fn();
    instance.onInputKeyDown(keyCode);
    expect(instance.onAddCommand).toHaveBeenCalledTimes(0);
  });
});
