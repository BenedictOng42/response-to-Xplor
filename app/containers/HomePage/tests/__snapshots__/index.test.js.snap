// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render correctly 1`] = `
<div>
  <InputWrapper>
    <WithStyles(Input)
      inputProps={
        Object {
          "aria-label": "Description",
        }
      }
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Input Command"
      value=""
    />
    <InputButtonWrapper>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Add Command
      </WithStyles(Button)>
    </InputButtonWrapper>
  </InputWrapper>
  <CommandWrapper>
    <WithStyles(Grid)
      alignItems="stretch"
      container={true}
      justify="space-evenly"
      spacing={16}
    >
      <CardWrapper>
        <WithStyles(CardContent)>
          <WithStyles(Typography)
            color="textPrimary"
            gutterBottom={true}
          >
            Valid Commands
          </WithStyles(Typography)>
        </WithStyles(CardContent)>
      </CardWrapper>
      <CardWrapper>
        <WithStyles(CardContent)>
          <WithStyles(Typography)
            color="textPrimary"
            gutterBottom={true}
          >
            Failed Commands
          </WithStyles(Typography)>
        </WithStyles(CardContent)>
      </CardWrapper>
    </WithStyles(Grid)>
    <h3>
      Output Result
    </h3>
    <TextField
      disabled={true}
      id="outlined-name"
      margin="normal"
      required={false}
      select={false}
      value="Must use a valid REPORT command"
      variant="outlined"
    />
    <TextField
      disabled={true}
      id="outlined-textarea"
      label="Output History"
      margin="normal"
      multiline={true}
      placeholder="Placeholder"
      required={false}
      select={false}
      variant="outlined"
    />
  </CommandWrapper>
</div>
`;
